### PR TITLE
refactor: add printer for metadata output

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
This PR adds a printer so user-specified cmd output in go scripting works as expected.

### CODE REVIEW

File: cmd/oras/internal/display/metadata/view/printer.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+type Outputable interface {
+	// WithOutput resets output of the handler.
+	WithOutput(out io.Writer)
+}
+
+// Printer prints.
+type Printer interface {
+	Printf(format string, a ...any) (n int, err error)
+	Println(a ...any) (n int, err error)
+	PrintJSON(object any) error
```
Comment (TerryHowe) [inline cmd/oras/internal/display/metadata/view/printer.go:37]: I was thinking the printer wouldn't know about format, but I'll have to think about that more. 

I was in the process of unwinding `verbose` everywhere which is a lot more complicated than this.

Comment (qweeah) [reply cmd/oras/internal/display/metadata/view/printer.go:37]: `verbose` serves only for **status output**. E.g. in the below output:
```console
➜  output oras push localhost:5000/demo:output image.tar --no-tty
Uploading 8b138a688d47 image.tar
Uploaded  8b138a688d47 image.tar
Pushed [registry] localhost:5000/demo:output
Digest: sha256:b7593bbc81769aa9fef0b37b73405eccfaf968a56e6b542614cfb19fe376e782
```
The status output is below and only covers **named** content
```console
Uploading 8b138a688d47 image.tar
Uploaded  8b138a688d47 image.tar
```

If we run it in verbose mode, then the output is
```console
➜  output oras push localhost:5000/demo:output image.tar --no-tty --verbose
Preparing image.tar
Uploading 8b138a688d47 image.tar
Uploading 44136fa355b3 application/vnd.oci.empty.v1+json
Uploaded  8b138a688d47 image.tar
Uploaded  44136fa355b3 application/vnd.oci.empty.v1+json
Uploading 262c1a7d7850 application/vnd.oci.image.manifest.v1+json
Uploaded  262c1a7d7850 application/vnd.oci.image.manifest.v1+json
Pushed [registry] localhost:5000/demo:output
Digest: sha256:262c1a7d7850ee622859898dafa7a13618fabb9beb5b4fa5dfb8cc1e73bbb8d7
```
In this case, also unnamed content is output:
```console
Uploading 8b138a688d47 image.tar
Uploading 44136fa355b3 application/vnd.oci.empty.v1+json
Uploaded  8b138a688d47 image.tar
Uploaded  44136fa355b3 application/vnd.oci.empty.v1+json
Uploading 262c1a7d7850 application/vnd.oci.image.manifest.v1+json
Uploaded  262c1a7d7850 application/vnd.oci.image.manifest.v1+json
```

Comment (qweeah) [reply cmd/oras/internal/display/metadata/view/printer.go:37]: This PR only focus on **metadata output**, in the above case, the metadata output is:
```console
Pushed [registry] localhost:5000/demo:output
Digest: sha256:262c1a7d7850ee622859898dafa7a13618fabb9beb5b4fa5dfb8cc1e73bbb8d7
```

and 

```console
Pushed [registry] localhost:5000/demo:output
Digest: sha256:b7593bbc81769aa9fef0b37b73405eccfaf968a56e6b542614cfb19fe376e782
```

**Comment by qweeah** [comment]
Discussed offline with @shizhMSFT, closing it and will refactor in more straightforward way.

**Pull request closed without merging**